### PR TITLE
AO3-6644 Adding jpeg support for collection header

### PR DESCRIPTION
**Pull request by JonathanS223**
# Pull Request Checklist

* [X] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [X] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [X] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
* [X] Have you added the [Jira](https://otwarchive.atlassian.net) issue number
  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [X] Do you have fewer than 5 pull requests already open? If not, please wait
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6644 (Please fill in issue number and remove this comment.)

## Purpose

Added jpeg support in the validates_format_of section of the :header_image_url

## Testing Instructions

Attempt to upload a jpeg formatted image. 

## References
N/A

## Credit

Ivedonestranger (he/him)

Jira: Jonathan S.

### CODE REVIEW

File: app/models/collection.rb [hunk 1]
```diff
@@ -156,7 +156,7 @@
     too_long: ts("must be less than %{max} characters long.", max: ArchiveConfig.SUMMARY_MAX)
 
   validates_format_of :header_image_url, allow_blank: true, with: URI::regexp(%w(http https)), message: ts("is not a valid URL.")
-  validates_format_of :header_image_url, allow_blank: true, with: /\.(png|gif|jpg)$/, message: ts("can only point to a gif, jpg, or png file."), multiline: true
+  validates_format_of :header_image_url, allow_blank: true, with: /\.(png|gif|jpg|jpeg)$/, message: ts("can only point to a gif, jpg, jpeg or png file."), multiline: true
```
[the remark this answers is not in the document]
Comment (JonathanS223) [reply app/models/collection.rb:159]: Huh. That's a new one. I'll be looking into this.

File: app/models/collection.rb [hunk 2]
```diff
@@ -156,7 +156,7 @@ def title
...
```
Comment (brianjaustin) [inline app/models/collection.rb:159]: I'm wondering if it makes sense to have both jpg and jpeg in the message, since they're the same file format. Maybe `jpg (jpeg)` or `jpg/jpeg`? The regex can also be simplified a bit:
```suggestion
  validates_format_of :header_image_url, allow_blank: true, with: /\.(png|gif|jpe?g)$/, message: ts("can only point to a gif, jpg, jpeg or png file."), multiline: true
```

File: app/models/collection.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (brianjaustin) [reply app/models/collection.rb:159]: If you haven't had a chance to dig in, https://stackoverflow.com/a/577675 has some good information (I had it handy because we made a similar change in https://github.com/otwcode/otwarchive/pull/4601).

TLDR - if someone changes the input to be multiline (like a textarea) they could, in theory enter something like
```
sneaky.jpeg
haha, another line
```
IIRC in most languages, `$` and `\z` end up being functionally equivalent, but not in Ruby

[the remark this answers is not in the document]
Comment (JonathanS223) [reply app/models/collection.rb:159]: Thanks! I haven't had the chance thanks to life. I appreciate the information to work on this!

File: app/models/collection.rb (hunk 2 above)
Comment (JonathanS223) [reply app/models/collection.rb:159]: Let me try that out. regex is still a beast that I'm attempting to master.

**Comment by JonathanS223** [comment]
Still working on this one. I should have some updates to the pull request this week.

**Comment by Bilka2** [comment]
I adopted this PR, updated code here: #5312
